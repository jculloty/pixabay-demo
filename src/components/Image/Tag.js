import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

import withApi from '../API';

class ImageTag extends PureComponent {
  static propTypes = {
    tag: PropTypes.string.isRequired,
    history: PropTypes.object.isRequired,
  }

  queryTag = () => {
    this.props.queryApi(this.props.tag);
    this.props.history.push('/');
  }

  render() {
    return (
      <span className="image-tag" key={this.props.tag} onClick={this.queryTag}>
        {this.props.tag}
      </span>
    );
  }
}

export default withRouter(withApi(ImageTag, ['queryApi']));
