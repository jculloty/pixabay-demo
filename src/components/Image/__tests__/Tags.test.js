import React from 'react';
import PropTypes, { shape } from 'prop-types';
import Enzyme, { shallow, mount, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import { MemoryRouter, BrowserRouter } from 'react-router-dom';

import Tag from '../Tag';

Enzyme.configure({ adapter: new Adapter() });

const buildComponent = function(tag = 'testing') {
  return (
    <MemoryRouter>
      <Tag tag={tag} />
    </MemoryRouter>
  );
};

const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
};

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
});

function mountWrap(node) {
  return mount(node, createContext());
}

describe('Image: Tag', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(buildComponent())
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Write more tests', () => {
    expect(true).toBe(false);
  });
});
