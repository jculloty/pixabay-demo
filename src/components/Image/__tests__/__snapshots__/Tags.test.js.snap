// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image: Tag renders correctly 1`] = `
<span
  className="image-tag"
  onClick={[Function]}
>
  testing
</span>
`;
